### ISSUE TITLE
postgresql@15 formula yields inconsistent paths

### ISSUE THREAD

**Issue by maparent** [Homebrew/homebrew-core#113705]
### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
https://gist.github.com/d15d394657beca0ea2ec32257be247ef
Note I am aware of the brew doctor warnings, and they do not create issues with postgresql@14.
```

### Verification

- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.

### What were you trying to do (and why)?

I was trying to install a postgresql extension (pgvector) on postgresql@15.

### What happened (include all command output)?

The output of `/opt/homebrew/opt/postgresql@15/bin/pg_config --includedir-server` is `/opt/homebrew/opt/postgresql@15/include/server` which does not match the actual location of include files, namely `/opt/homebrew/opt/postgresql@15/include/postgresql/server/`.
This prevents building many postgresql extensions.
Meanwhile, `/opt/homebrew/opt/postgresql@15/bin/pg_config --includedir` accurately gives `/opt/homebrew/opt/postgresql@15/include`, but I think this is a poor choice as it implies the Postgres include files are linked directly in the `/opt/homebrew/include` directory, instead of being namespaced.
Even /opt/homebrew/opt/postgresql@15/include/postgresql/server/ means there is no namespacing.
I recommend using the same patterns as postgresql@14, where all directories are correctly namespaced.

### What did you expect to happen?

`pg_config` should give accurate information; moreover the full package name should be part of the paths.

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
brew install postgresql@15
/opt/homebrew/opt/postgresql@15/bin/pg_config --includedir-server
ls /opt/homebrew/opt/postgresql@15/include/postgresql/server/
```

**Comment by ankane** [comment]
Hey @maparent, the `--includedir-server` path issue was fixed in #114321, so you should be able to install extensions now (outside of Homebrew).

**Comment by maparent** [comment]
I can confirm that the information given by `pg_config` is now accurate, and it is now possible to build extensions. Thank you for the changes, closing the bug.
I would still suggest that the base include directory should be namespaced by version number to facilitate parallel installs, but that is a separate issue.
Regards
